parity-clib/examples/cpp/main.cpp

```diff
@@ -14,172 +14,134 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
 
+#ifdef __clang__
+#pragma clang diagnostic error "-Wvexing-parse"
+#endif
 #include <chrono>
+#include <cstdint>
 #include <parity.h>
+#include <parity.hpp>
 #include <regex>
 #include <string>
 #include <thread>
+namespace {
+using namespace std::literals::string_literals;
+using parity::ethereum::ParityEthereum;
+using parity::ethereum::parity_subscription;
 
-void* parity_run(std::vector<const char*>);
-int parity_subscribe_to_websocket(void*);
-int parity_rpc_queries(void*);
+void parity_subscribe_to_websocket(ParityEthereum &ethereum);
+void parity_rpc_queries(ParityEthereum &);
+parity::ethereum::ParityEthereum parity_run(const std::vector<std::string> &);
 
 const int SUBSCRIPTION_ID_LEN = 18;
 const size_t TIMEOUT_ONE_MIN_AS_MILLIS = 60 * 1000;
 const unsigned int CALLBACK_RPC = 1;
 const unsigned int CALLBACK_WS = 2;
 
 struct Callback {
-	unsigned int type;
-	long unsigned int counter;
+  size_t type;
+  std::uint64_t counter;
+  void operator()(const std::string_view response) {
+    if (type == CALLBACK_RPC) {
+      counter -= 1;
+    } else if (type == CALLBACK_WS) {
+      std::match_results<std::string_view::iterator> results;
+      std::regex is_subscription(
+          "\\{\"jsonrpc\":\"2.0\",\"result\":\"0[xX][a-fA-"
+          "F0-9]{16}\",\"id\":1\\}");
+      if (std::regex_match(response.begin(), response.end(), results, is_subscription)) {
+        counter -= 1;
+      }
+    } else {
+      abort();
+    }
+  }
 };
 
 // list of rpc queries
-const std::vector<std::string> rpc_queries {
-	"{\"method\":\"parity_versionInfo\",\"params\":[],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_getTransactionReceipt\",\"params\":[\"0x444172bef57ad978655171a8af2cfd89baa02a97fcb773067aef7794d6913fff\"],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_estimateGas\",\"params\":[{\"from\":\"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5\"}],\"id\":1,\"jsonrpc\":\"2.0\"}",
-	"{\"method\":\"eth_getBalance\",\"params\":[\"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5\"],\"id\":1,\"jsonrpc\":\"2.0\"}"
+const std::vector<std::string> rpc_queries{
+    "{\"method\":\"parity_versionInfo\",\"params\":[],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
+    "{\"method\":\"eth_getTransactionReceipt\",\"params\":[\"0x444172bef57ad978655171a8af2cfd89baa02a97fcb773067aef7794d6913fff\"],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
+    "{\"method\":\"eth_estimateGas\",\"params\":[{\"from\":\"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5\"}],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
+    "{\"method\":\"eth_getBalance\",\"params\":[\"0x0066Dc48bb833d2B59f730F33952B3c29fE926F5\"],\"id\":1,\"jsonrpc\":\"2.0\"}"s,
```

Comment:
All of these would be cleaner with https://www.geeksforgeeks.org/raw-string-literal-c/.

Reply:
Also could we maybe make them `constexpr std::vector<std::string_view>` ?